spec/test_app_templates/lib/generators/test_app_generator.rb

```diff
@@ -5,6 +5,67 @@ class TestAppGenerator < Rails::Generators::Base
   # so the following path gets us to /path/to/hyrax/spec/test_app_templates/
   source_root File.expand_path('../../../../spec/test_app_templates/', __FILE__)
 
+  def conditional_postgresql_for_ci_environment
+    # For now, postgresql is something we are looking at adding to CircleCI
```

Comment:
could all of this be handled by `DATABASE_URL` instead? https://edgeguides.rubyonrails.org/configuring.html#configuring-a-database

Reply:
I think we could. I also believe we could then run postgresql OR sqlite locally for specs based on an ENV.  That would be dreamy.

Reply:
I did some further digging, and those ENV's currently come from CircleCI. Also, based on my read of that configuring a database, you'd be able to use DATABASE_URL for local development.

Perhaps this is something to bump to @rotated8  and samvera's orbs?

Reply:
@jeremyf i think you can set environment variables for a job here, without requiring changes to the orb: https://github.com/samvera/hyrax/blob/master/.circleci/config.yml#L20-L22

i think adding a line like `DATABASE_URL: 'postgresql://hyrax_user:hyrax_password@postgres/hyrax?pool=5'` might do the trick?

Reply:
@no-reply I'll give it a go on my local machine; I'd love to remove some of that conditional logic.

Reply:
@no-reply I'm struggling to unpack this one. In part, what is in place currently works for CI. What I'm proposing is that we merge this, and then I can tinker with refining the CI build process to try the DATABASE_URL.

The iteration on the build process is rather slow locally. I think having an already building Postgresql CI environment would be helpful as I work to fine tune that build process.

All of that said, I could be doing this work on a branch of this branch.

Reply:
I have confirmed that if I provide a `DATABASE_URL=postgresql://localhost:5432/internal_development?pool=5 bundle exec rails console` on my machine that this works. I have not attempted a username/password.

Reply:
I'm writing these as notes, as I explore: This is where I'm currently getting the ENVs (https://github.com/samvera-labs/samvera-circleci-orb/blob/master/src/executors/ruby_fcrepo_solr_redis_postgres.yml#L41-L43). I believe I would need to adjust our `.circleci/config.yml`

Reply:
I switched my config to use a different adapter and database name.

I then ran `DATABASE_URL=postgresql://localhost:5432/internal_development?pool=5 bundle exec rails console` in the .internal_test_app. The environment used the DATABASE_URL.

Next consideration, is can I take the CircleCI environment variables and craft a new environment variable?

Reply:
possibly it's okay to just hard code the CI config in multiple ENV variables as a first step, then refactor the duplication out after?

Reply:
I think I'm hearing that it'd be nice if the orb exported a DATABASE_URL variable. So, samvera-labs/samvera-circleci-orb#42

Reply:
@rotated8 thanks, I'll get to hard-coding, and see about remediating that particular file. I'm a big fan of how small this PR is becoming.

Reply:
The last set of commits and rebasing addresses the DATABASE_URL, and opts for the hard-coding as talked about in https://wiki.lyrasis.org/display/samvera/Samvera+Tech+Call+2020-07-15

It's much cleaner and removes several switches (though pushes people to locally use postgresql when they regen their internal test app).

Reply:
oh wow this looks great! thanks for your documentation.

i appreciate the personal nature of your code comments. :)